scene/gui/tree.cpp

```diff
@@ -1793,9 +1797,13 @@ void Tree::draw_item_rect(TreeItem::Cell &p_cell, const Rect2i &p_rect, const Co
 	Size2 ts = p_cell.text_buf->get_size();
 	bool rtl = is_layout_rtl();
 
-	int w = 0;
+	rect.set_size(rect.get_size() - Size2i(theme_cache.inner_item_margin_right, theme_cache.inner_item_margin_bottom));
```

Comment:
Because of RTL that's not necessary.
If we do account for all margin values we get a weird behavior where increasing margin left acts as a margin right increase

Reply:
> Because of RTL that's not necessary.
> If we do account for all margin values we get a weird behavior where increasing margin left acts as a margin right increase

I'd say this indicates that some calculations down below are conceptually incorrect. :upside_down_face:

---

BTW there's `Rect2i::grow_individual` so it could be used instead of manually setting position/size:
```cpp
Rect2i rect = p_rect.grow_individual(-theme_cache.inner_item_margin_left, -theme_cache.inner_item_margin_top, -theme_cache.inner_item_margin_right, -theme_cache.inner_item_margin_bottom);
```

Reply:
Thanks @kleonc, updated it to include your suggestion